### PR TITLE
Controller Rotation Test

### PR DESCRIPTION
**Pull request by smrghsh**
Changes:
- Including the [Quaternion.js](https://github.com/infusion/Quaternion.js/) library as a dependency.
- Exposing the rotation as Euler angles (while respecting the `angleMode`) for progress for #158. There is some gimbal-locking type behavior that is persisting.
- Updating the manual input test to show controller tracking with orientation as well as primary and second button presses
- Added notice about gimbal locking to documentation

https://user-images.githubusercontent.com/22751315/186528906-cf60441d-1cab-431e-b950-fecd9eb41ce2.mp4

### CODE REVIEW

File: src/p5xr/core/p5xrInput.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ export default class p5xrInput {
       vec3.transformMat3(this._dir, this._dir, normalMat);
     }
     this.direction = this._dir;
+    // console.log(p5.instance.angleMode);
```
Comment (stalgiag) [inline src/p5xr/core/p5xrInput.js:30]: leftover

File: src/p5xr/core/p5xrInput.js [hunk 2]
```diff
@@ -63,6 +64,18 @@ export default class p5xrInput {
...
     return new p5.Vector(p.x, p.y, p.z);
   }
+  /** @returns {p5.Vector} Returns the current rotation as a Vector */
+  get rotation() {
+    this.updatePose();
+    if (this._pose){
+      const {x, y, z, w} = this._pose?.transform?.orientation;
+      let q = new Quaternion(x, y, z, w);
+      let e = q.toEuler();
+      let c = (p5.instance.angleMode == RADIANS)? 1 : 180 / Math.PI; // TODO: Change output with ANGLE MODE
```
Comment (stalgiag) [inline src/p5xr/core/p5xrInput.js:74]: Better to use the private p5 helper [methods](https://github.com/processing/p5.js/blob/main/src/math/trigonometry.js#L345-L360) or [constants ](https://github.com/processing/p5.js/blob/main/src/core/constants.js#L184)for this.

Comment (stalgiag) [inline src/p5xr/core/p5xrInput.js:74]: Make sure to run `npm run grunt` locally to check for linting issues before push. Here you are using the `isLooselyEqual` comparison operator vs the `isStrictlyEqual` one which is against the linting rules. [Here is a reference on the difference.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness)

File: src/p5xr/core/p5xrInput.js [hunk 3]
```diff
@@ -63,6 +64,18 @@ export default class p5xrInput {
     const p = this._pose?.transform?.position;
     return new p5.Vector(p.x, p.y, p.z);
   }
+  /** @returns {p5.Vector} Returns the current rotation as a Vector */
```
Comment (stalgiag) [inline src/p5xr/core/p5xrInput.js:67]: A good idea to mention the gimbal locking behavior here and provide `applyMatrix(p5xrInput.pose)` as an alternative.

**Review by stalgiag — CHANGES_REQUESTED** [review]

File: src/p5xr/core/p5xrInput.js (hunk 1 above)
Comment (smrghsh) [reply src/p5xr/core/p5xrInput.js]: removed

File: src/p5xr/core/p5xrInput.js (hunk 2 above)
Comment (smrghsh) [reply src/p5xr/core/p5xrInput.js]: This function now checks for `angleMode `and uses p5's angle converter. I tested by using the manual test for input and changing the `angleMode `during set up.

Comment (smrghsh) [reply src/p5xr/core/p5xrInput.js]: Thank you!

**Review by stalgiag — APPROVED** [review]

**Comment by stalgiag** [comment]
Forgot to do this!

@all-contributors please add @smrghsh for code, design, and documentation

**Pull request merged**